### ISSUE TITLE
LT01: tsql: Alias with missing space and `=` is fiixed in correctly, code corruption

### ISSUE THREAD

**Issue by fmms** [sqlfluff/sqlfluff#4619]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I tried to format

```
SELECT col1 =GETDATE()
```

### Expected Behaviour

Statement gets formatted and ` ` is introduced after the `=`

### Observed Behaviour

The statement is fixed to `SELECT AS col1 = GETDATE()``which is not correct sql at all.

### How to reproduce

```
PS > cat .\aliasEquals.sql
SELECT col1 =GETDATE()
PS > sqlfluff fix --dialect tsql .\aliasEquals.sql --force
==== finding fixable violations ====
== [aliasEquals.sql] FAIL
L:   1 | P:  14 | LT01 | Expected single whitespace between raw comparison
                       | operator '=' and function name identifier.
                       | [layout.spacing]
==== fixing violations ====
1 fixable linting violations found
FORCE MODE: Attempting fixes...
Persisting Changes...
== [aliasEquals.sql] PASS
Done. Please check your files to confirm.
  [1 unfixable linting violations found]
PS > cat .\aliasEquals.sql
SELECT AS col1 = GETDATE()
```

### Dialect

tsql

### Version

```
==== sqlfluff ====
sqlfluff:                2.0.2 python:                 3.11.0
implementation:        cpython verbosity:                   2
dialect:                  tsql templater:                 raw
rules:                                all
```

### Configuration

none

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
With the latest `sqlfluff` this statement is properly linted and formatted.

```bash
 sqlfluff format test.sql -d tsql

==== finding fixable violations ====
== [test.sql] FAIL                                                                                                                                                                
L:   1 | P:  14 | LT01 | Expected single whitespace between raw comparison
                       | operator '=' and function name identifier.
                       | [layout.spacing]
== [test.sql] FIXED
```

And the result:

```sql

SELECT col1 = GETDATE()

```

**Comment by peterbud** [comment]
Could one of the maintainer close this issue pls? I don't have access to do that. @alanmcruickshank @WittierDinosaur ?
